numpy/core/numeric.py

```diff
@@ -1343,11 +1343,15 @@ def roll(a, shift, axis=None):
     ----------
     a : array_like
         Input array.
-    shift : int
-        The number of places by which elements are shifted.
-    axis : int, optional
-        The axis along which elements are shifted.  By default, the array
-        is flattened before shifting, after which the original
+    shift : int or tuple of ints
+        The number of places by which elements are shifted.  If a tuple,
+        then `axis` must be a tuple of the same size, and each of the
+        given axes is shifted by the corresponding number.  If an int
```

Comment:
So to be clear. If you don't mind just ignoring the broadcasting for now, I think I would prefer something like:
```
try:
     axes = tuple(axis)
except:
    axes = (axis,)
```
or the inverted `try: operator.index(axis)`. But if the broadcasting is important to you, maybe as is, is just simpler, hmmm.